js/src/dropdown.js

```diff
@@ -263,7 +263,7 @@ class Dropdown extends BaseComponent {
       typeof config.reference.getBoundingClientRect !== 'function'
     ) {
       // Popper virtual elements require a getBoundingClientRect method
-      throw new Error(`${NAME}: Option "reference" provided type "object" without a required "getBoundingClientRect" method.`)
+      throw new TypeError(`${NAME}: Option "reference" provided type "object" without a required "getBoundingClientRect" method.`)
```

Comment:
I wonder if we should make the related tests stricter, probably makes sense.

Reply:
Updated.
Also, changed the `NAME` to upper case to make the consistency.
Should check for the same in `jQueryInterface` also? :thinking:

Reply:
What do you mean?

Reply:
I mean should the tests be updated here? (To check if the `error` is the instance of `TypeError`)
https://github.com/twbs/bootstrap/blob/7e6d7b8f817ad1212238f15124335d37c93ebfea/js/tests/unit/dropdown.spec.js#L1560-L1564

Reply:
Ah, yeah let's switch any other instances to `toThrowError`